CRM/Mailing/BAO/Mailing.php

```diff
@@ -1997,12 +1997,16 @@ public static function &report($id, $skipDetails = FALSE, $isSMS = FALSE) {
         $row['bounce_rate'] = (100.0 * $mailing->bounce) / $mailing->queue;
         $row['unsubscribe_rate'] = (100.0 * $row['unsubscribe']) / $mailing->queue;
         $row['optout_rate'] = (100.0 * $row['optout']) / $mailing->queue;
+        $row['opened_rate'] = ($row['opened'] / $mailing->delivered * 100.0);
```

Comment:
We should probably add protection against a hard error if delivered === 0 (presumably a very obscure situation but worth handling I think)

Reply:
yeh i guess so

Reply:
Also - can we stdise bracketting - I prefer but mostly just thing same is better 
```
 $row['opened_rate'] = ($row['opened'] / $mailing->delivered) * 100;
```
with conditional

 $row['opened_rate'] = $mailing->delivered ?  (($row['opened'] / $mailing->delivered) * 100) : '';